### PR TITLE
Logic split

### PR DESCRIPTION
**Pull request by zaqbez39me**

### CODE REVIEW

File: main/algorithm.py [hunk 1]
```diff
@@ -5,9 +5,8 @@
 from sys import platform
 from zipfile import ZipFile
 
-from django.http import HttpResponse
-
 import env_consts as ec
+from django.http import HttpResponse
```
Comment (osx11) [inline main/algorithm.py:9]: According to PEP8, importing of (or from) third-party libraries should be done secondly. https://peps.python.org/pep-0008/#imports

Suggestion: rollback the change

File: main/linux.py [hunk 2]
```diff
@@ -18,11 +17,14 @@ def install_libre():
     ec.INSTALLED_LIBRE = True
 
 
-def convert_linux(filepath: str, files, converted_file_path: str, has_type_in_request):
+def convert_linux(
+    filepath: str, files, converted_file_path: str, has_type_in_request: dict
```
Comment (osx11) [inline main/linux.py:21]: type of files argument is not defined

File: main/linux.py [hunk 3]
```diff
@@ -46,7 +48,7 @@ def convert_linux(filepath: str, files, converted_file_path: str, has_type_in_re
 
     command = f"cd {filepath}"
     for type_in in has_type_in_request:
-        if has_type_in_request[type_in] and type_in is not ".pdf":
+        if has_type_in_request[type_in] and type_in != ".pdf":
```
Comment (osx11) [inline main/linux.py:51]: `is` checks for object identity, whereas `!=` checks for value identity. Here, probably, you should check for value identity.

Suggestion: rollback changes

**Review by osx11 — CHANGES_REQUESTED** [review]
Please, view my comments for `main/algorithm.py`

File: main/linux.py (hunk 3 above)
Comment (osx11) [reply main/linux.py:51]: My mistake. It is ok.

File: main/linux.py (hunk 2 above)
Comment (osx11) [reply main/linux.py:21]: Also, type of has_type_in_request should be replaced with Dict[..., ...] **if possible**, where `...` is types of key and value respectively

File: main/algorithm.py (hunk 1 above)
Comment (zaqbez39me) [reply main/algorithm.py:9]: I am not sure that after such change linter tests will pass

Comment (osx11) [reply main/algorithm.py:9]: I am not sure we should use a tool that goes against the principles of the language. Ok, if there is no way to overcome it, I will mark this one as resolved.

File: main/windows.py [hunk 4]
```diff
@@ -60,7 +62,7 @@ def windows_convert_xlsx(excel, xlsx_filepath: str, pdf_filepath: str):
     xl_sheets.Close(True)
 
 
-def windows(paths, files, has_type_in_request: dict):
+def windows(paths, files: List[str], has_type_in_request: Dict[str, bool]):
```
Comment (osx11) [inline main/windows.py:65]: type of `paths` is not specified

**Review by osx11 — CHANGES_REQUESTED** [review]

**Review by osx11 — APPROVED** [review]

**Pull request merged**